### ISSUE TITLE
Issue 05 — `Categorical` gradient & Fisher can silently become `NaN` for large natural parameters

### ISSUE THREAD

**Issue by docxology** [ReactiveBayes/ExponentialFamily.jl#294]
# Issue 05 — `Categorical` gradient & Fisher can silently become `NaN` for large natural parameters

**Component:** `src/distributions/categorical.jl` (getgradlogpartition 94-105, getfisherinformation 107-129)
**Severity:** MAJOR (silent NaN at extreme params)
**Type:** Numerical robustness

## Summary
`Categorical` computes its log-partition with the numerically stable `logsumexp`
(line 91), but its gradient and Fisher use raw `exp(η)`:

```julia
getgradlogpartition(::NaturalParametersSpace, ::Type{Categorical}, conditioner) = (η) -> begin
    ...
    sumη = mapreduce(exp, +, η)          # ← overflow for large η
    return map(d -> exp(d) / sumη, η)
end
```
For large natural parameters (large log-probabilities), `exp(η)` overflows to `Inf`
and the gradient becomes `NaN`/`Inf·0`, while `getlogpartition` remains fine.

## Evidence / repro (Julia 1.12.6)
```julia
η = [800.0, 810.0, 0.0]
getlogpartition(η)        # => 810.00004   (stable, logsumexp)
getgradlogpartition(η)    # => [NaN, NaN, 0.0]
getfisherinformation(η)   # => NaN entries
```

## Impact
Silently corrupts the EF gradient/Fisher for `Categorical` (and any downstream
`ReactiveMP` VMP/EP update that uses these) when natural parameters are large. In
normal use logits are modest so this is latent, but it is a genuine numerical
correctness hazard and inconsistent with the stable log-partition.

## Proposed fix
Use the softmax / `logsumexp` form:
```julia
getgradlogpartition(..., conditioner) = (η) -> begin
    (conditioner === length(η)) || throw(DimensionMismatch(...))
    return softmax(η)                    # numerically stable
end
```
computing the Fisher from the softmax probabilities:
```julia
p = softmax(η)
I = Diagonal(p) - p * p'
```

## Notes
`getlogpartition` (line 82-92) is already correct via `logsumexp`; the fix only needs
to make the gradient/Fisher consistent with it.

**Comment by bvdmitri** [comment]
Fixed in #303 and released in v2.6.0 (see the `[2.6.0]` CHANGELOG entry): the natural-space gradient and Fisher information for `Categorical` are now computed via `softmax`, so they no longer become `NaN` for large natural parameters (`src/distributions/categorical.jl`). Regression test added in 2f1eb8d.
